phpBB/phpbb/lock/posting.php

```diff
@@ -67,29 +64,14 @@ public function acquire(int $creation_time, string $form_token): bool
 	{
 		$this->set_lock_name($creation_time, $form_token);
 
-		// Lock is held for session, cannot acquire it
-		if ($this->cache->_exists($this->lock_name))
+		// Lock is held for session, cannot acquire it unless special flag for testing is set
+		if ($this->cache->_exists($this->lock_name) && !$this->config->offsetExists('ci_tests_no_lock_posting'))
```

Comment:
Having tests stuff in the core looks hackish a bit tbh.
Could it be something like `&& PHPBB_ENVIRONMENT !== 'test'` or `&& PHPBB_ENVIRONMENT === 'production'`?

Reply:
It is somewhat of a hack but otherwise it also wouldn't be possible to test the expected behavior during production use.